MIGRATIONGUIDE.md

```diff
@@ -0,0 +1,363 @@
+ change log section below.
+ 
+However, because this client library does not provide explicit enum for certain fields, your 
+integration relying on string equality of will not be caught by the compiler. If the enum value 
+is removed or replaced with a different name, your code execution flow can get affected. 
+Furthermore, the upgrade can change semantics of certain enums, and you should verify the 
+implications on using the same enum values. For example, in [version change](https://stripe.com/docs/upgrades#2017-06-05),
+`under_review` now takes the meaning previously overloaded in `other` value for 
+`Account#Verification#getDisabledReason`; if you use `other` to determine if the disabled account is
+ under review, you should check for value `under_review` instead.
+ 
+```java
+Account account = Account.retrieve();
+// need to change "other" to "under_review" instead
+if (account.getVerification().disabledReason.equals("other")) {
+  // code path assuming account is still under review
+}
+```
+
+Similar to enum is the value of `object` field in most of model classes. In [version 
+change](https://stripe.com/docs/upgrades#2018-09-24) the value for `File` is changed from 
+`file_upload` to `file`. Explicit check for `File#getObject` should be updated accordingly.
+
+Beyond the contract that fully-typed system can offer is the change in business implication, 
+where the schema remains the same. This is less common, but does happen. For example in [version 
+change](https://stripe.com/docs/upgrades#2018-05-21), an `id` of a subscription in line item lost 
+its meaning as the subscription ID.  
+ 
+#### API requests 
+ 
+On the API requests, because this client library takes an untyped map as a parameter request to an 
+API method, eg. `Source#create(Map<String, Object> params)`, removed parameters due to API 
+upgrade will not be caught statically. For example, before [version change](https://stripe.com/docs/upgrades#2018-08-23)
+`Subscription#cancel` allows deferred cancellation with `at_period_end` request parameter, but 
+now removed in favor of `Subscription#update` with `cancel_at_period_end` parameter. Similarly, 
+there is extensive change in parameters for `PaymentIntent#create` and `PaymentIntent#confirm` 
+for beta users of `PaymentIntent` in [version change](https://stripe.com/docs/upgrades#2018-11-08)
+
+Parameter validation is another common change. For example, 
+ [version change](https://stripe.com/docs/upgrades#2018-10-31) requires `Customer#update` 
+ parameter `name` to be within 250 character-long. On the other hand, parameter validation can 
+ also become less restrictive, and the assumption you had on the valid returned response can be 
+ violated. For example, if you had relied on Stripe validation on uniqueness of SKU `attributes` during 
+ creation/update, and had built business logic around this assumption, after [version change](https://stripe.com/docs/upgrades#2018-09-06)
```

Comment:
`after [version change](https://stripe.com/docs/upgrades#2018-09-06)` -> `after the [2018-09-06 version change](https://stripe.com/docs/upgrades#2018-09-06)`